atlasdb-client/src/main/java/com/palantir/atlasdb/transaction/impl/ReadTransaction.java

```diff
@@ -127,7 +126,7 @@ public Transaction delegate() {
 
     private void checkTableName(TableReference tableRef) {
         SweepStrategy sweepStrategy = sweepStrategies.get(tableRef);
-        if (sweepStrategy == SweepStrategy.THOROUGH) {
+        if (sweepStrategy.mustCheckImmutableLockAfterReads()) {
             throw new SafeIllegalStateException(
                     "Cannot read from a table with a thorough sweep strategy in a read only transaction.");
         }
```

Comment:
Could we change the message to something like - "Can only read from a table with a no/conservative sweep strategy in a read only transaction." ?

Reply:
done